File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        string stack;

        for (char c : s) {
            if (c == '*') {
                stack.pop_back();  // remove the closest non-star to the left
            } else {
                stack.push_back(c);  // add non-star character
            }
        }

        return stack;
    }
};

File: 2470-removing-stars-from-a-string/removing-stars-from-a-string.py
class Solution:
    def removeStars(self, s: str) -> str:
        stack = []

        for char in s:
            if char == '*':
                stack.pop()  # remove the closest non-star to the left
            else:
                stack.append(char)  # add non-star character

        return ''.join(stack)